app/Keycloak/RealmCollection.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Keycloak\Collection;
+
+use App\Keycloak\Realm;
+use Illuminate\Support\Collection;
+
+/**
+ * @extends Collection<int, Realm>
+ */
+final class RealmCollection extends Collection
```

Comment:
Another naming pattern can be to just use the plural form `Realms`, but no need to change it. (This plural form follows the approach of not suffixing classes with Interface or Exception)